lib/workers/repository/extract/extract-fingerprint-config.ts

```diff
@@ -0,0 +1,56 @@
+import type { RenovateConfig } from '../../../config/types';
+import { get, getManagerList } from '../../../modules/manager';
+import type { WorkerExtractConfig } from '../../types';
+
+export interface FingerprintExtractConfig
+  extends Omit<WorkerExtractConfig, 'manager' | 'fileList'> {
+  manager?: string;
+  fileList?: string[];
+}
+
+export function extractFingerprintConfig(
+  config: RenovateConfig
+): FingerprintExtractConfig[] {
+  const managerExtractConfigs: FingerprintExtractConfig[] = [
+    ...((config.regexManagers ?? []) as FingerprintExtractConfig[]),
+  ];
+
+  let managerList = getManagerList();
+  const { enabledManagers } = config;
+  if (enabledManagers?.length) {
+    managerList = managerList.filter((manager) =>
+      enabledManagers.includes(manager)
+    );
+  }
+
+  for (const manager of managerList) {
+    const managerConfig: WorkerExtractConfig = config[manager] as any;
+    const language = get(manager, 'language');
+    const languageConfig = (
+      language ? (config[language] ? config[language] : {}) : {}
+    ) as FingerprintExtractConfig;
+    const filteredConfig = {} as FingerprintExtractConfig;
+
+    // npmrc and npmrcMerge
+    if (manager === 'npm') {
```

Comment:
I don't really like this hardcoding of manager logic. From a fingerprinting point of view, would it really make any difference?

Reply:
Nope just that it will be undefined for other managers

Reply:
Better to simplify and have that than to hardcode manager logic within lib/workers